src/module_3/push_exploration.md

```diff
@@ -0,0 +1,5364 @@
+plt.ylabel('Precision')
+plt.title('Precision-Recall Curve')
+
+plt.tight_layout()
+plt.show()
+```
+
+
+    
+![png](push_exploration_files/push_exploration_75_0.png)
+    
+
+
+
+```python
+X = final_data.drop(columns='outcome')
+y_val_pred = model.predict(X_val)
+
+# Print the classification report for validation results
+print("Classification Report:")
+print(classification_report(y_val, y_val_pred))
+
+print("Confusion Matrix:")
+cm = confusion_matrix(y_val, y_val_pred)
+print(cm)
+
+coefficients = model.coef_[0]  # Coefficients for the first (and only) class
+coefficients_df = pd.DataFrame({'Feature': X.columns, 'Coefficient': coefficients})
+coefficients_df = coefficients_df.reindex(coefficients_df['Coefficient'].abs().sort_values(ascending=False).index)
+coefficients_df.tail(10)
+```
+
+    Classification Report:
+                  precision    recall  f1-score   support
+    
+               0       0.99      0.78      0.88    213182
+               1       0.04      0.61      0.08      3213
+    
+        accuracy                           0.78    216395
+       macro avg       0.52      0.70      0.48    216395
+    weighted avg       0.98      0.78      0.86    216395
+    
+    Confusion Matrix:
+    [[167075  46107]
+     [  1247   1966]]
+
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>Feature</th>
+      <th>Coefficient</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>43</th>
+      <td>product_facialskincare</td>
+      <td>0.018726</td>
+    </tr>
+    <tr>
+      <th>54</th>
+      <td>product_juicesquash</td>
+      <td>0.016255</td>
+    </tr>
+    <tr>
+      <th>7</th>
+      <td>count_pets</td>
+      <td>0.015768</td>
+    </tr>
+    <tr>
+      <th>16</th>
+      <td>day_of_week</td>
+      <td>0.009763</td>
+    </tr>
+    <tr>
+      <th>8</th>
+      <td>people_ex_baby</td>
+      <td>-0.009313</td>
+    </tr>
+    <tr>
+      <th>19</th>
+      <td>product_babyfood12months</td>
+      <td>-0.003423</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>order_id</td>
+      <td>-0.002904</td>
+    </tr>
+    <tr>
+      <th>52</th>
+      <td>product_householdsundries</td>
+      <td>-0.002543</td>
+    </tr>
+    <tr>
+      <th>45</th>
+      <td>product_femininecare</td>
+      <td>-0.001950</td>
+    </tr>
+    <tr>
+      <th>6</th>
+      <td>global_popularity</td>
+      <td>0.000000</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+## Reiterate on feature selection
+
+Based on the best model we achieved, which is using Logistic Regression with l2 and class_weight balanced, we will take at the final weights used (most and least) and do another round of feature selection
```

Comment:
All the models seemed very similar, where did you conclude that this particular model was better than the other ones?